### PR TITLE
batch retransmit_to in retransmit_stage

### PR DESCRIPTION
**Pull request by jbiseda**
#### Problem
Computation interleaved with packet transmission makes measuring computation vs transmission time more difficult.

#### Summary of Changes
Consolidate packet retransmission in retransmit_stage.

./gce.sh create -p testnet-dev-jbiseda -n 10 -c 1
./net.sh start -c "bench-tps=1=--tx_count 5000 --thread-batch-sleep-ms 250"

changes left, master right:
Screen Shot retransmit_stage_batch-master

### CODE REVIEW

**Review by behzadnouri — COMMENTED** [review]
I am not sure if you saw the ongoing discord discussions in `#network-protocols` channel regarding packets drops: https://github.com/solana-labs/solana/issues/19079
https://discord.com/channels/428295358100013066/478692221441409024/870707709639548989

It might be the case that excessive batching is actually causing higher bursts and overall more packet drops; and we might need to in fact go in the other direction and implement some rate-limiting, back-pressure, etc.

So it will be very helpful if with such changes we can get some before/after measurements of the impact and ensure that it does in fact improve things:
* There are already some `RetransmitStats` metrics in this code:
   https://github.com/solana-labs/solana/blob/c7b986220/core/src/retransmit_stage.rs#L64-L79
   https://github.com/solana-labs/solana/blob/c7b986220/core/src/retransmit_stage.rs#L135-L188
   We can start a validator on testnet running the current master code, leave it run for a while, then restart it with this change included and compare metrics; similar to: https://github.com/solana-labs/solana/pull/14781#issuecomment-766391155 
   * This would not lead any metrics on packet drops though.
* For packet drops, one (imperfect) proxy are the repair-metrics. We can start a GCE cluster of maybe 10 nodes or so, with and without this code and compare repair metrics similar to: https://github.com/solana-labs/solana/pull/14910#issuecomment-805107406

**Comment by CriesofCarrots** [comment]
I'll defer to Behzad on the review (I'm in the `git blame` here because I moved a bunch of stuff to `solana-gossip`; haven't worked in this area much otherwise.)

File: core/src/retransmit_stage.rs [hunk 1]
```diff
@@ -294,31 +294,25 @@ fn retransmit(
...
-        );
+            ClusterInfo::get_retransmit_to_dests(&neighbors[1..], true, socket_addr_space)
+        } else {
+            vec![]
+        };
+        let child_addrs: Vec<_> =
+            ClusterInfo::get_retransmit_to_dests(&children, !anchor_node, socket_addr_space);
+        dest_addrs.extend(child_addrs);
+        compute_turbine_peers.stop();
+        compute_turbine_peers_total += compute_turbine_peers.as_us();
```
Comment (behzadnouri) [inline core/src/retransmit_stage.rs:312]: why change this metric?

**Comment by behzadnouri** [comment]
Thank you for adding the metrics plots. Can you please describe the cluster set-up you used to obtain the metrics? and, what is your conclusion as the impact of change?

Also, it would be great if:
* use the "Problem/Summary of Changes" template for pull-request description: https://github.com/solana-labs/solana/blob/master/CONTRIBUTING.md#the-pr-problem-statement
* include before/after metrics in the same plot similar to:
  https://github.com/solana-labs/solana/pull/14910#issuecomment-805107406
  right now even the axes are not the same; which makes the comparison more difficult. e.g. 0-100 vs 0-600.

**Comment by jbiseda** [comment]
Closing per test results

**Pull request closed without merging**